src/stash/src/memory.rs

```diff
@@ -40,21 +40,13 @@ impl<S: Stash> Memory<S> {
         }
     }
 
-    async fn consolidate_collection<K, V>(
-        &mut self,
-        collection: StashCollection<K, V>,
-    ) -> Result<(), StashError>
-    where
-        K: Data,
-        V: Data,
-    {
-        let since = self.since(collection).await?;
-        self.consolidate_id(&collection.id, since);
-        Ok(())
-    }
-
-    fn consolidate_id(&mut self, collection_id: &Id, since: Antichain<Timestamp>) {
-        if let Some(entry) = self.entries.get_mut(collection_id) {
+    fn consolidate_id(&mut self, collection_id: Id) {
+        if let Some(entry) = self.entries.get_mut(&collection_id) {
+            let since = match self.sinces.get(&collection_id) {
+                Some(since) => since,
+                // If we don't know the since for this collection, don't consolidate.
+                None => return,
```

Comment:
Why not remove the collection ID like we do down below?

Reply:
Down below handles when the since is known, but empty (end of time).

Reply:
Hmm yeah I guess this is safer. It'd remove the little cheat we have to do in the test, and would also prevent situations where the memory cache and on disk present  *slightly* differently in hopefully meaningless ways, but I've already written some cache bugs here so...let's be safe instead. Ok, changing to remove. Thanks.